File: backend/src/main/java/me/performancereservation/domain/review/Review.java
package me.performancereservation.domain.review;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import me.performancereservation.domain.common.BaseEntity;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Review extends BaseEntity {
    //리뷰 아이디, 유저 아이디, 공연 아이디, 회차, 리뷰 내용

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long performanceId;

    private Long userId;

    @Column(nullable = false, length = 1000)
    private String comments; //또는 text 형식?

    @Builder
    public Review(Long userId, Long performanceId, String comments) {
        this.performanceId = performanceId;
        this.userId = userId;
        this.comments = comments;
    }

    public void updateComments(String comments) {
        this.comments = comments;
    }
}

File: backend/src/main/kotlin/me/performancereservation/domain/review/Review.kt
package me.performancereservation.domain.review

import jakarta.persistence.*
import me.performancereservation.domain.common.BaseEntity
import me.performancereservation.domain.user.entitiy.User

@Entity
data class Review(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    val performanceId: Long,

    val userId: Long,

    @Column(nullable = false, length = 1000)
    var comments: String
) : BaseEntity() {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is User) return false
        if (id == null || other.id == null) return false
        return id == other.id && updatedAt == other.updatedAt
    }

    override fun hashCode(): Int {
        var result = id?.hashCode() ?: 0
        result = 31 * result + updatedAt.hashCode()
        return result
    }

    fun updateComments(comments: String) {
        this.comments = comments
    }
}